### DISCUSSION TITLE
Submit Jobs to YarraServer from Linux Fileserver

### DISCUSSION THREAD

**Discussion by liamtimms** [yarra-framework/yarra-framework.github.io#4] [General]
Hi,

First, I wanted to say thank you for the previous advice for running the yarraserver on Ubuntu 22.04. I'm now wondering what advice you might have regarding running the Yarra SAC client (or equivalent) on a Linux machine as well. Our current workflow uses the RDS client to transfer raw TWIX data to our Linux based backup systems from various scanners. We aim to then submit jobs from these raw data storage Linux systems to the yarraserver for reconstruction. Since there is not a Windows machine involved after the RDS client, what is the best way to submit jobs to the yarraserver? 

Thank you for your time and attention.

**Comment by tblock79** [comment]
Hi Liam,

We normally push cases to our Yarra servers from the scanners using the ORT client. If we want to retrospectively process cases, we usually use the SAC client for this. However, if you don't have access to Windows, there are multiple options:

- We have developed the "Yarra WebClient", which is a web-based version of the SAC client that allows users to submit cases from the browser. It runs under Linux and also includes an interface to Yarra Archive Search. At our center, our researchers use it to search for cases in our archive and then to push those cases to a research Yarra server. It is available in this Git archive: https://bitbucket.org/yarra-dev/yarra-webclient. Unfortunately, the installation is a bit difficult and there is no documentation yet (that's why we never made any announcements about it). The installation definitely requires some knowledge of Linux / Docker / Python.

- The other alternative would be that you write a small script to submit cases to YarraServer. YarraServer has an incoming "queue" folder  and it will process cases when it finds a .dat Twix file and matching .task file (with same filename beside the extension). The .task file is a small text file in .ini format with meta information that the SAC/ORT client creates -- it basically tells YarraServer which recon mode to execute. Thus, you could write a script that first copies the Twix .dat file into the queue folder and then creates the .task file (you need to do it in that order to prevent YarraServer from starting the reconstruction while the .dat file is still being copied). The format of the .task file is pretty simple -- you could just use the SAC client to generate a sample file to see which entries are needed, or look at the code of the function "bool ortReconTask::generateTaskFile()" here: https://bitbucket.org/yarra-dev/yarraclient/src/master/OfflineReconClient/ort_recontask.cpp.

- A third option could be to compile the SAC client for Linux or Mac (depending on what you need). Since it has been written with the Qt framework, it should compile on other operating systems as well.

Hope this helps!

Best,
  Tobias
